### PR TITLE
Add a navigation toggle and vue-slideout

### PR DESCRIPTION
**Pull request by ChristophWurst**
There is one glitch when the menu nav is hidden again for the first time by clicking the icon: it shows blank space instead of moving the content to the right. @skjnldsv could you have a look and work your magic? :)

### CODE REVIEW

**Comment by ChristophWurst** [comment]
@skjnldsv what's the status here?

**Comment by skjnldsv** [comment]
@ChristophWurst needs to find a fix for the server compatibility.
I would be in favour of NOT initiating it if the toggle is present OR if app-content is not here.

**Comment by ChristophWurst** [comment]
You mean with https://github.com/nextcloud/server/pull/14972?

**Comment by skjnldsv** [comment]
Yep :)

File: src/components/AppContent/AppContent.vue [hunk 1]
```diff
@@ -21,11 +21,72 @@
...
+	},
+	mounted() {
+		this.mc = new Hammer(this.$el)
+		this.mc.on('swipeleft swiperight', e => {
+			this.handleSwipe(e)
+		})
+	},
+	unmounted() {
+		this.mc.off('swipeleft swiperight')
+		this.ms.destroy()
```
Comment (juliushaertl) [inline src/components/AppContent/AppContent.vue]:
```suggestion
		this.mc.destroy()
```

Comment (skjnldsv) [reply src/components/AppContent/AppContent.vue]: Thanks!! Nice catch!

**Pull request merged**